src/main/java/org/molgenis/data/annotation/reportrvcf/FOR.java

```diff
@@ -20,119 +19,137 @@
-
-        HashMap<String, Integer> countPerGeneLeftover = new HashMap<String, Integer>();
-
-        for (String variant : variantToGene.keySet()) {
-            String gene = variantToGene.get(variant);
-
-            if (countPerGeneLeftover.containsKey(gene)) {
-                countPerGeneLeftover.put(gene, countPerGeneLeftover.get(gene) + 1);
-            } else {
-                countPerGeneLeftover.put(gene, 1);
-            }
-        }
-
-        System.out.println("left over variant counts per gene: " + countPerGeneLeftover.toString());
-
-
-        PrintWriter pw = new PrintWriter(genesFORoutput);
-        pw.println("Gene" + "\t" + "Expected" + "\t" + "Observed" + "\t" + "MissedFrac");
-        for (String gene : countPerGeneExpected.keySet()) {
-            //if no leftovers, subtract 0
-            int subtract = 0;
-            if (countPerGeneLeftover.containsKey(gene)) {
-                //  System.out.println("countPerGeneLeftover.containsKey(gene) " + gene);
-                subtract = countPerGeneLeftover.get(gene);
-            }
-            int exp = countPerGeneExpected.get(gene);
-            int obs = (countPerGeneExpected.get(gene) - subtract);
-            pw.println(gene + "\t" + exp + "\t" + obs + "\t" + ((1.0 - ((double) obs / (double) exp))));
-        }
-        pw.flush();
-        pw.close();
-
-    }
+public class FOR
+{
+
+	private File originalVcfFile;
+	private File rvcfFile;
+	private File genesFORoutput;
+
+	/*
+	args[0] = working dir (e.g. /Users/joeri/Desktop/GAVIN-APP/1000G_diag_FDR/FOR/)
+	args[1] = original VCF (e.g. GAVIN_FOR_benchmark_goldstandard_nodup_gonl.vcf)
+	args[2] = RVCF output (e.g. RVCF_GAVIN_FOR_benchmark_goldstandard_nodup_gonl_r1.0.vcf)
+	args[3] = FOR summary per gene, output file (e.g. FOR.tsv)
+	 */
+	public static void main(String[] args) throws Exception
+	{
+		String dir = args[0];
+		FOR falseOM = new FOR(new File(dir + args[1]), new File(dir + args[2]), new File(dir + args[3]));
+		falseOM.go();
+	}
+
+	public FOR(File originalVcfFile, File rvcfFile, File genesFORoutput) throws Exception
+	{
+		this.originalVcfFile = originalVcfFile;
+		this.rvcfFile = rvcfFile;
+		this.genesFORoutput = genesFORoutput;
+
+	}
+
+	public void go() throws Exception
+	{
+		//first, since we don't know the original, read the input VCF file that was processed before
+		//and count how many pathogenic variants per gene should have been found
+		//assumes that all variants in this list are (likely) pathogenic
+		VcfReader vcf = GavinUtils.getVcfReader(originalVcfFile);
+		Iterator<VcfRecord> originalVcfIterator = vcf.iterator();
+
+		HashMap<String, String> variantToGene = new HashMap<String, String>(); //e.g. 10_126092389_G_A -> OAT, 10_126097170_C_T -> OAT
+		while (originalVcfIterator.hasNext())
+		{
+			GavinRecord record = new GavinRecord(originalVcfIterator.next());
+
+			String gene;
+			// TODO check if : split is correct
```

Comment:
@dennishendriksen I think this TODO is yours, can you check this, or do we need @joerivandervelde for this?